pkg/controllers/workgenerator/controller.go

```diff
@@ -618,6 +625,8 @@ func setBindingStatus(works map[string]*fleetv1beta1.Work, resourceBinding *flee
 	if appliedCond.Status == metav1.ConditionTrue {
 		availableCond = buildAllWorkAvailableCondition(works, resourceBinding)
 		resourceBinding.SetConditions(availableCond)
+	} else if appliedCond.Status == metav1.ConditionFalse {
+		resourceBinding.RemoveConditions(resourceBinding.Status.Conditions, string(fleetv1beta1.ResourceBindingAvailable))
```

Comment:
why do we need this? We already set the availableCondition to false when the resource changes.
Also, this checks the appliedCondition but remove the AvailableCondition, is that by design?

Reply:
When the work is updated, we change the applied condition to false with the message that the work is synchronizing. Without this, the condition will stay as that when the applied is false.

Reply:
Will update test cases for the integration test to follow instead.